### PR TITLE
[MRG] Add analysis tab

### PR DESCRIPTION
**Pull request by chenghuzi**

### CODE REVIEW

**Comment by jasmainak** [comment]
The diff is still too large ... what's going on?

**Comment by chenghuzi** [comment]
> The diff is still too large ... what's going on?

some reorganization of the code. Not really a lot of changes.

**Comment by jasmainak** [comment]
don't mix things in one PR ... code re-organization separate PR, then features in separate PR

File: hnn_core/gui/gui.py [hunk 1]
```diff
@@ -164,12 +463,42 @@ def __init__(self, theme_color="#8A2BE2", log_window_height="100px",
                                                  value='Joblib',
                                                  description='Backend:')
 
+        # analysis tab
+        analysis_style = {'description_width': '200px'}
+        self.widget_if_save_figures = Dropdown(options=[('Yes', True),
+                                                        ('No', False)],
+                                               value=False,
+                                               description='Save figures:',
+                                               style=analysis_style)
```
Comment (jasmainak) [inline hnn_core/gui/gui.py:472]: let's get rid of this widget ... we don't need it

File: hnn_core/gui/gui.py [hunk 2]
```diff
@@ -164,12 +463,42 @@ def __init__(self, theme_color="#8A2BE2", log_window_height="100px",
...
+        self.widget_if_save_spectral_data = Dropdown(
+            options=[('Yes', True), ('No', False)], value=False,
+            description='Save Spectral Data:', style=analysis_style)
+
```
Comment (jasmainak) [inline hnn_core/gui/gui.py:476]: same here ... don't need

File: hnn_core/gui/gui.py [hunk 3]
```diff
@@ -164,12 +463,42 @@ def __init__(self, theme_color="#8A2BE2", log_window_height="100px",
...
+                                               disabled=False,
+                                               style=analysis_style)
+        self.widget_dipole_smooth = FloatText(
+            value=30.0,
+            description='Dipole Smooth Window (ms):',
+            disabled=False, style=analysis_style)
+        self.widget_if_record_somatic_v = Dropdown(
+            options=[('Yes', True), ('No', False)], value=False,
+            description='Record Somatic Voltages:', style=analysis_style)
+
```
Comment (jasmainak) [inline hnn_core/gui/gui.py:491]: This doesn't seem to be related to "analysis" ... you need to specify at the beginning of the simulation

File: hnn_core/gui/gui.py [hunk 4]
```diff
@@ -477,10 +838,16 @@ def run_notebook_cells(self):
...
+        function cellContainsInitOrMarkdown(idx){
+            const cell = Jupyter.notebook.get_cell(idx);
+            if(cell.cell_type!=='code'){
+                return true;
+            }
+            else{
+                const textVal = cell.element[0].childNodes[0].textContent;
+                return textVal.includes('HNNGUI()') || textVal.includes(
+                    'HNNGUI');
+            }
```
Comment (jasmainak) [inline hnn_core/gui/gui.py:850]: separate PR please!

File: hnn_core/gui/gui.py (hunk 3 above)
Comment (chenghuzi) [reply hnn_core/gui/gui.py:491]: This is from the old gui. I can move it

**Comment by chenghuzi** [comment]
@jasmainak  could you help to review once we pass the tests?

File: hnn_core/gui/gui.py [hunk 5]
```diff
@@ -28,79 +29,101 @@
                         VBox, link)
 from ipywidgets.embed import embed_minimal_html
 
+_spectrogram_color_maps = [
+    "jet",
```
Comment (jasmainak) [inline hnn_core/gui/gui.py:33]: can we remove the "jet" colormap? It's not a good colormap for power calculation. Power is a positive quantity for which diverging colormap is not recommended.

File: hnn_core/gui/gui.py [hunk 6]
```diff
@@ -590,18 +661,13 @@ def _run_button_clicked(b):
...
+                self.connectivity_widgets, self.analysis_config, b)
 
-        def _handle_viz_layout_change(layout_option):
-            return _initialize_viz_window(
-                self._viz_grid,
-                self._visualization_window, self.simulation_data,
-                self.plot_outputs_list, self.plot_dropdowns_list,
-                self.layout['visualization_window'],
-                layout_option=layout_option.new)
+        def _handle_viz_layout_change(_):
```
Comment (jasmainak) [inline hnn_core/gui/gui.py:668]: what does the underscore do when passed to a function? I have never seen this. Is it a new Python feature or existed before?

**Comment by jasmainak** [comment]
some minor comments, otherwise LGTM. Address them asap and I'll merge!

File: hnn_core/gui/gui.py (hunk 6 above)
Comment (chenghuzi) [reply hnn_core/gui/gui.py:668]: It does nothing basially, more like a placeholder. I use it as `_` is not treated as a sort of "variable" that should be accessed somewhere in vscode so the color won't change.  https://stackoverflow.com/questions/5893163/what-is-the-purpose-of-the-single-underscore-variable-in-python

**Comment by jasmainak** [comment]
Thanks @chenghuzi ! go, go, go 💪 💪 💪
